MonoGame.Framework/Graphics/GraphicsDebug.DirectX.cs

```diff
@@ -0,0 +1,77 @@
+﻿using SharpDX.Direct3D11;
+using System.Collections.Generic;
+
+namespace Microsoft.Xna.Framework.Graphics
+{
+    public partial class GraphicsDebug
+    {
+        private readonly GraphicsDevice _device;
+        private readonly InfoQueue _infoQueue;
+        private readonly Queue<GraphicsDebugMessage> _cachedMessages;
+
+        public GraphicsDebug(GraphicsDevice device)
+        {
+            _device = device;
+            _infoQueue = _device._d3dDevice.QueryInterfaceOrNull<InfoQueue>();
+            _cachedMessages = new Queue<GraphicsDebugMessage>();
+
+            _infoQueue.PushEmptyRetrievalFilter();
+            _infoQueue.PushEmptyStorageFilter();
+        }
+
+        public bool TryDequeueMessage(out GraphicsDebugMessage message)
```

Comment:
Maybe define this function in the GraphicsDebug class and make this `PlatformTryDequeueMessage` like we do for other methods with platform specific features. Or include a file with a default implementation of this function that always returns false and include it for other platforms. That way this won't break compilation when users switch targets.